haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,182 @@
+        except AWSConfigurationError as e:
+            raise AmazonBedrockConfigurationError(message=e.message) from e
+        except Exception as e:
+            raise AmazonBedrockConfigurationError(
+                "Could not connect to Amazon Bedrock. Make sure the AWS environment is configured correctly."
+            ) from e
+
+        model_available = model_name_or_path in available_model_ids
+        if not model_available:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} is not available in Amazon Bedrock. "
+                f"Please make sure the model is available in the configured AWS region and you've been granted access."
+            )
+
+        return model_supported
+
+    def _prepare_invoke(self, prompt, **kwargs):
+        del kwargs["top_k"]
+        if self.model_name_or_path in ["amazon.titan-text-express-v1", "amazon.titan-text-lite-v1"]:
+            kwargs["stopSequences"] = kwargs["stop_words"] or []
+            kwargs["topP"] = 1 if "topP" not in kwargs else kwargs["topP"]
+            kwargs["temperature"] = 0.3 if "temperature" not in kwargs else kwargs["temperature"]
+            kwargs["maxTokenCount"] = self.max_length
+            del kwargs["stop_words"]
+            body = json.dumps({"inputText": prompt, "textGenerationConfig": {**kwargs}})
+        if self.model_name_or_path in ["ai21.j2-ultra-v1", "ai21.j2-mid-v1"]:
+            kwargs["topP"] = 1 if "topP" not in kwargs else kwargs["topP"]
+            kwargs["temperature"] = 0.3 if "temperature" not in kwargs else kwargs["temperature"]
+            kwargs["maxTokens"] = self.max_length
+            del kwargs["stop_words"]
+            body = json.dumps({"prompt": prompt, **kwargs})
+        if self.model_name_or_path in ["cohere.command-text-v14"]:
+            kwargs["temperature"] = 0.3 if "temperature" not in kwargs else kwargs["temperature"]
+            kwargs["max_tokens"] = self.max_length
+            del kwargs["stop_words"]
+            body = json.dumps({"prompt": prompt, **kwargs})
+        if self.model_name_or_path in ["anthropic.claude-v1", "anthropic.claude-v2", "anthropic.claude-instant-v1"]:
+            kwargs["max_tokens_to_sample"] = self.max_length
+            kwargs["temperature"] = 0 if "temperature" not in kwargs else kwargs["temperature"]
+            kwargs["top_k"] = 250 if "top_k" not in kwargs else kwargs["top_k"]
+            kwargs["top_p"] = 1 if "top_p" not in kwargs else kwargs["top_p"]
+            kwargs["stop_sequences"] = ["\n\nHuman:"] if "stop_sequences" not in kwargs else kwargs["stop_sequences"]
+            del kwargs["stop_words"]
+            body = json.dumps({"prompt": f"\n\nHuman: {prompt}\n\nAssistant:", **kwargs})
+        return body
+
+    def extract_responses(self, r):
+        out = json.loads(r["body"].read().decode())
+        return out
```

Comment:
I guess we don't need a method for this. It's sufficient to just call it once in `invoke`. We also should explicitly specify utf-8 decoding. Feel free to look up how it's done in other invocation layers, e.g. under https://github.com/viveksilimkhan1/haystack/blob/244732413379571f99c0fe19a3401d34bc5fd17e/haystack/nodes/prompt/invocation_layer/sagemaker_meta.py#L231